lib/firespring_dev_commands/templates/git.rb

```diff
@@ -159,6 +159,49 @@ def create_merge_task!
           end
         end
       end
+
+      # Create the rake task for the git commit status task.
+      def create_commit_status_task!
+        # Have to set a local variable to be accessible inside of the instance_eval block
+        exclude = @exclude
+
+        DEV_COMMANDS_TOP_LEVEL.instance_eval do
+          namespace :git do
+            return if exclude.include?(:commit_status)
+
+            # TODO: Clean, comments and description
+            desc 'Add status to commit' \
+                 "\n\tuse TODO"
+
+            task :create_commit_status do
+              # Key Values
+              repository = ENV['REPOSITORY'].to_s.strip
+              branch = ENV['BRANCH'].to_s.strip
+              status = ENV['STATUS'].to_s.strip
+
+              raise 'Repository name is required' unless repository
+              raise 'Branch name is required' unless branch
+              raise 'Status is required' unless status
+
+              # Validate status
+              valid_statuses = %w(error failure pending success)
+              raise "Invalid status: #{status}. Valid statuses are: #{valid_statuses.join(', ')}" unless valid_statuses.include?(status)
+
+              # Optional Values
+              context = ENV['CONTEXT'].to_s.strip
```

Comment:
Should _any_ of these be required? Not sure exactly what happens when you exclude them.